File: 2150-find-all-lonely-numbers-in-the-array/2150-find-all-lonely-numbers-in-the-array.py
class Solution:
    def findLonely(self, nums: List[int]) -> List[int]:
      print(nums)
      nums.sort()
      nums = [-2] + nums + [1000002]
      n=len(nums)
      lonelyNums = []
      for i in range(1,n-1):
        if nums[i-1] < nums[i]-1 and nums[i+1] > nums[i]+1:
          lonelyNums += [nums[i]]
      return lonelyNums

File: 2150-find-all-lonely-numbers-in-the-array/2150-find-all-lonely-numbers-in-the-array.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& nums) 
    {
      long freq[1000002] = {0};
      for(int i:nums)
      {
        freq[i]++;
      }
      vector<int> lonelyNumbers;
      for(int i:nums)
      {
        if(freq[i]==1 && (i == 0 || freq[i-1] == 0) && freq[i+1] == 0)
        {
          lonelyNumbers.push_back(i);
        }
      }
      return lonelyNumbers;
    }
};